.ado/windows-vs-pr.yml

```diff
@@ -365,15 +365,21 @@ jobs:
         X64Debug:
           BuildConfiguration: Debug
           BuildPlatform: x64
-        X86Debug:
-          BuildConfiguration: Debug
-          BuildPlatform: x86
         X64Release:
           BuildConfiguration: Release
           BuildPlatform: x64
+        X86Debug:
+          BuildConfiguration: Debug
+          BuildPlatform: x86
         X86Release:
           BuildConfiguration: Release
           BuildPlatform: x86
+        ARM64Debug:
+          BuildConfiguration: Debug
```

Comment:
Do these both really need to be added to the PR pipeline? For that matter, why does the desktop PR pipeline have so many configs?

UWP gets x86 Debug and x64 Release - the PR CI is supposed to be a lean check.

Reply:
I was not aware UWP was missing build combinations. Why is this?

Maybe for UWP, heavily based on COM and obviously WinRT, compile/link issues are less likely to happen.

For raw Windows Desktop, we have found issues as specific as `x86|Debug|v141`.

In general, I think we should build every flavor.
Total time should be the same, thanks to parallel builds.

The checks should be lean, but coverage should be wide.

Reply:
Sorry, short answer: Yes. Both will be added.

Reply:
@JunielKatarn please don't add all combinations, see the comment above this:
```
      matrix: # Why we only build some flavors: https://github.com/microsoft/react-native-windows/issues/4308
```
https://github.com/microsoft/react-native-windows/issues/4308

Reply:
I didn't see that comment. It's not in the Desktop job:
```
  - job: RNWDesktopPR
    displayName: Desktop PR
    dependsOn: Setup
    condition: ne( dependencies.Setup.outputs['checkPayload.shouldSkipPRBuild'], 'True' )
    strategy:
      matrix:
        X64Debug:
          BuildConfiguration: Debug
...
```

I can adjust this in a subsequent change, but we do build the whole current matrix for Desktop.
I'll look at that commented issue to see the what was the agreed exclusion criteria.